readthedocs/rtd_tests/tests/test_api.py

```diff
@@ -1345,6 +1345,56 @@ def test_github_skip_signature_validation(self, trigger_build):
         )
         self.assertEqual(resp.status_code, 200)
 
+    def test_github_dont_trigger_double_sync(self, trigger_build):
+        """Don't trigger a sync twice if the webhook has the create/delete events."""
+        integration = Integration.objects.create(
+            project=self.project,
+            integration_type=Integration.GITHUB_WEBHOOK,
+            provider_data={
+                'events': [
+                    GITHUB_CREATE,
+                    GITHUB_DELETE,
+                ],
+            },
+            secret=None,
+        )
+
+        client = APIClient()
+
+        headers = {
+            GITHUB_EVENT_HEADER: GITHUB_PUSH,
+        }
+        payload = {
+            'ref': 'master',
+            'created': True,
+            'deleted': False,
+        }
+        resp = client.post(
+            reverse(
+                'api_webhook_github',
+                kwargs={'project_slug': self.project.slug}
+            ),
+            payload,
+            format='json',
+            **headers
+        )
+        self.assertFalse(resp.json()['versions_synced'])
+
+        headers = {
+            GITHUB_EVENT_HEADER: GITHUB_CREATE,
```

Comment:
I think we are missing a test case here, when all of these are met:

* `GITHUB_PUSH` event
* `created=True` 
* `GITHUB_CREATE` *not in* `Integration.provider_data.events`

Reading the code it seems that should do the sync versions, right?

Reply:
Added